rocketpy/Function.py

```diff
@@ -600,8 +635,8 @@ def setDiscreteBasedOnModel(self, modelFunction, oneByOne=True):
             Zs = np.array(self.getValue(mesh))
             self.setSource(np.concatenate(([Xs], [Ys], [Zs])).transpose())
 
-        self.setInterpolation(modelFunction.__interpolation__)
-        self.setExtrapolation(modelFunction.__extrapolation__)
+        self.setInterpolation(self.__interpolation__)
+        self.setExtrapolation(self.__extrapolation__)
```

Comment:
I don't remember discussing this, but I may have forgotten.

From what I see, we have to different behaviors:
 - Keeping interpolation and extrapolation unchanged when discretizing.
 - Using the interpolation and extrapolation of the model Function.

Could you point out the pros and cons of each? How would the first option work in case self originally has a callable source?

Feel free to keep the proposed solution if you find it the best option. However, make sure to document this behavior.

Reply:
Both ways are possible, however, during coding with this method, I found that keeping the `self` properties is usually more useful:

- In general, the Model Function was used only to ensure the same discretization as other in order to perform arithmetic operations. Therefore, the main point was to make the `xArray` equally discretized;
- Due the fact that the arithmetic operations now no longer require the same interpolation for array computations, it seems unnecessary to force `self` into the same interpolation of the model Function. That may cause issues if an "exact" extrapolation (such as linear (i.e. natural)) is substituted by an "inaccurate" extrapolation (such as polynomial)".

As an example of the latter, the IdentityMap is based on accurate linear extrapolation, is one discretizes it based on a model Function of another kind (e.g. something really common) great inaccuracies may appear.